src/vfs/syspath_win32.c

```diff
@@ -0,0 +1,280 @@
+        char *pattern = strjoin(node->syspath.path, "\\*.*", NULL);
+        wchar_t *wpattern = WIN_UTF8ToString(pattern);
+        free(pattern);
+        search_handle = FindFirstFile(wpattern, &fdata);
+        free(wpattern);
+
+        if(search_handle == INVALID_HANDLE_VALUE) {
+            vfs_set_error_win32();
+            return NULL;
+        }
+
+        idata = calloc(1, sizeof(VFSWin32IterData));
+        idata->search_handle = search_handle;
+        idata->last_result = WIN_StringToUTF8(fdata.cFileName);
+        *opaque = idata;
+
+        if(!strcmp(idata->last_result, ".") || !strcmp(idata->last_result, "..")) {
+            return vfs_syspath_iter(node, opaque);
+        }
+
+        return idata->last_result;
+    }
+
+    idata = *opaque;
+    free(idata->last_result);
+    idata->last_result = NULL;
+
+    if(FindNextFile(idata->search_handle, &fdata)) {
+        idata->last_result = WIN_StringToUTF8(fdata.cFileName);
+
+        if(!strcmp(idata->last_result, ".") || !strcmp(idata->last_result, "..")) {
+            return vfs_syspath_iter(node, opaque);
+        }
+
+        return idata->last_result;
+    }
+
+    if(GetLastError() != ERROR_NO_MORE_FILES) {
+        vfs_set_error_win32();
+    }
+
+    return NULL;
+}
+
+static void vfs_syspath_iter_stop(VFSNode *node, void **opaque) {
+    VFSWin32IterData *idata = *opaque;
+
+    if(idata) {
+        if(!FindClose(idata->search_handle)) {
+            vfs_set_error_win32();
+        }
+
+        free(idata->last_result);
+        free(idata);
+    }
+}
+
+static char* vfs_syspath_repr(VFSNode *node) {
+    return strfmt("filesystem path (win32): %s", node->syspath.path);
+}
+
+static char* vfs_syspath_syspath(VFSNode *node) {
+    return strdup(node->syspath.path);
+}
+
+static bool vfs_syspath_mkdir(VFSNode *node, const char *subdir) {
+    if(!subdir) {
+        subdir = "";
+    }
+
+    char *p = strfmt("%s%c%s", node->syspath.path, '\\', subdir);
+    wchar_t *wp = WIN_UTF8ToString(p);
+    bool ok = CreateDirectory(wp, NULL);
+    DWORD err = GetLastError();
+
+    if(!ok && err == ERROR_ALREADY_EXISTS) {
+        VFSNode *n = vfs_locate(node, subdir);
+
+        if(n && vfs_query_node(n).is_dir) {
+            ok = true;
+        } else {
+            vfs_set_error("%s already exists, and is not a directory", p);
+        }
+
+        if(node != n) {
+            vfs_freetemp(n);
+        }
+    }
+
+    if(!ok) {
+        vfs_set_error("Can't create directory %s (win32 error: %lu)", p, err);
+    }
+
+    free(p);
+    free(wp);
+    return ok;
+}
+
+static VFSNodeFuncs vfs_funcs_syspath = {
+    .repr = vfs_syspath_repr,
+    .query = vfs_syspath_query,
+    .free = vfs_syspath_free,
+    .locate = vfs_syspath_locate,
+    .syspath = vfs_syspath_syspath,
+    .iter = vfs_syspath_iter,
+    .iter_stop = vfs_syspath_iter_stop,
+    .mkdir = vfs_syspath_mkdir,
+    .open = vfs_syspath_open,
+};
+
+static void vfs_syspath_normalize(char *path) {
```

Comment:
It absolutely isn't, I just hate ugly inconsistent paths in logs. If you're worried about potential bugs in this function, perhaps this should be done at the representation level (i.e. in `vfs_syspath_syspath`)?